cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -903,6 +1033,42 @@ fn insert_common_epilogue(
         pos.func.dfg.append_inst_arg(inst, csr_ret);
     }
 
+    if csrs.iter(FPR).len() != 0 {
+        let mut fpr_offset = 0;
+
+        // `stack_store` is not directly encodable in x86_64 at the moment, so we'll need a base
+        // address. We are well after postopt could run, so load the CSR region base once here,
+        // instead of hoping that the addr/store will be combined later.
+        let stack_addr = pos.ins().stack_addr(types::I64, *csr_slot, 0);
+
+        // At this point we won't be using volatile registers for anything except for return
+        // registers. Arbitrarily pick RBP as it won't hold an interesting value, and we're about
+        // to restore it at the end of the function anyway.
+        pos.func.locations[stack_addr] = ir::ValueLoc::Reg(RU::rbp as u16);
```

Comment:
This is brittle in the face of frame pointer elision. A more applicable register would be `r11`, but there are no rex encodings of `fstDisp8` or `fldDisp8`, so using `r11` as a base violates constraints that the registers must have index <=7.